src/state.ts

```diff
@@ -0,0 +1,69 @@
+import InstanceSkel = require('../../../instance_skel');
+import { Subject, Observable } from 'rxjs';
+import { takeUntil, filter } from 'rxjs/operators';
+import { CompanionFeedbackEvent } from '../../../instance_skel_types';
+import { UiConfig } from './config';
+
+export class UiFeedbackState {
+  private feedbackUnsubscribe$ = new Subject<string>();
+  private state: { [feedbackId: string]: any } = {};
+
+  constructor(private instance: InstanceSkel<UiConfig>) {}
+
+  /**
+   * Register a stream of feedback values from the mixer.
+   * Must be called from the feedback subscribe callback to connect the mixer state with companion.
+   * The stream will be subscribed until we call `unsubscribeFeedback` with the internal feedback ID.
+   * @param evt The feedback metadata from Companion
+   * @param stream$ The observable stream of values for this feedback
+   */
+  connect(evt: CompanionFeedbackEvent, stream$: Observable<any>): void {
+    const unsubscribe$ = this.feedbackUnsubscribe$.pipe(filter(fid => fid === evt.id || !fid));
```

Comment:
It looks like calling connect multiple times for the same observable and evt.type will cause checkFeedbacks() to be called multiple times for each change.
This wont be good for performance

Reply:
Good catch! I think, this will only happen if someone registers the exact same feedback source on multiple banks in parallel.

Do you have an idea how to solve this? Debouncing would delay the result …

Reply:
I could do a very short debounce, though, to catch event bursts, like 10 ms

Reply:
Yes, it would only be if using the same feedback source, but I dont think that unlikely. For example, I have the same mute button with feedback on 3 pages of my streamdeck (x32 not this module), which mutes my mic (for videocalls).

I would suggest tracking the active subscriptions, similar to what I have done here: https://github.com/bitfocus/companion-module-behringer-x32/blob/master/src/state.ts#L31
You will need some 'stream' id, but once you have that it will be easy to subscribe once and reference count the subscription

Reply:
I will put this into an issue and plan a rework after the major release!

Reply:
https://github.com/bitfocus/companion-module-soundcraft-ui/issues/18